nucliadb/src/nucliadb/ingest/fields/base.py

```diff
@@ -346,18 +346,22 @@ async def set_vectors(self, payload: ExtractedVectorsWrapper) -> Optional[Vector
             if len(payload.vectors.vectors.vectors) > 0:
                 actual_payload.vectors.CopyFrom(payload.vectors.vectors)
             await self.storage.upload_pb(sf, actual_payload)
-            self.extracted_vectors = actual_payload
+            self.extracted_vectors[vectorset] = actual_payload
         return vo
 
     async def get_vectors(
         self, vectorset: Optional[str] = None, force: bool = False
     ) -> Optional[VectorObject]:
-        if self.extracted_vectors is None or force:
+        # compat with vectorsets coming from protobuffers where no value is
+        # empty string instead of None. This shouldn't be handled here but we
+        # have to make sure it gets the correct vectorset
+        vectorset = vectorset or None
+        if self.extracted_vectors.get(vectorset, None) is None or force:
```

Comment:
`.get(vectorset, None)` is redundant, `.get(vectorset)` does the exact same thing. Same below.